### PR TITLE
Add JSON serialization, JSON schema, and validation

### PR DESCRIPTION
**Pull request by macumber**

### CODE REVIEW

File: spec/model_spec.rb [hunk 1]
```diff
@@ -439,9 +440,30 @@
...
     # install graphviz and make sure dot is in the path
     model.save_graphviz('shell.dot')
     system('dot shell.dot -Tpdf -o shell.pdf')
+
+    model.save("shell.json")
+    s = model.to_s
+    obj = JSON.parse(s, symbolize_names: true)
+    model2 = Topolys::Model.from_json(obj)
+    model2.save("shell2.json")
+    schema = Topolys::Model.schema
```
Comment (rd2) [inline spec/model_spec.rb:455]: Thanks for this ...

File: lib/topolys/schema/topolys.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (rd2) [inline lib/topolys/schema/topolys.json:1]: Why draft-04 (instead of 07)?

Comment (macumber) [reply lib/topolys/schema/topolys.json:1]: I tried 07 but couldn’t get visual studio code or the Json-schema gem to work with it :-/

Comment (rd2) [reply lib/topolys/schema/topolys.json:1]: Got it. But eventually moving up to 05, 06 or 07 will require some adaptation (e.g. $id, min/max constraints).

Comment (macumber) [reply lib/topolys/schema/topolys.json:1]: Yes hopefully we can upgrade, I just couldn’t get it to work last night

**Pull request merged**